tests/pkg_semtech-loramac/main.c

```diff
@@ -0,0 +1,362 @@
+/*
+ * Copyright (C) 2017 Inria
+ *               2017 Inria Chile
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     test
+ *
+ * @file
+ * @brief       Semtech LoRaMAC test application
+ *
+ * @author      Alexandre Abadie <alexandre.abadie@inria.fr>
+ * @author      Jose Alamos <jose.alamos@inria.cl>
+ */
+
+#include <string.h>
+
+#include "msg.h"
+#include "shell.h"
+
+#include "net/netdev.h"
+#include "net/loramac.h"
+#include "semtech_loramac.h"
+
+#include "sx127x.h"
+#include "sx127x_params.h"
+#include "sx127x_netdev.h"
+
+sx127x_t sx127x;
+static uint8_t rx_buf[242] = { 0 };
+static char print_buf[48] = { 0 };
+
+static void _hex_to_bytes(const char *hex, uint8_t *byte_array)
+{
+    const uint8_t charmap[] = {
+        0x00, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07, /* 01234567 */
+        0x08, 0x09, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, /* 89:;<=>? */
+        0x00, 0x0a, 0x0b, 0x0c, 0x0d, 0x0e, 0x0f, 0x00, /* @ABCDEFG */
+        0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, /* HIJKLMNO */
+    };
+
+    size_t len = strlen(hex);
+    for (uint8_t pos = 0; pos < len; pos += 2) {
+        uint8_t idx0 = ((uint8_t)hex[pos + 0] & 0x1F) ^ 0x10;
+        uint8_t idx1 = ((uint8_t)hex[pos + 1] & 0x1F) ^ 0x10;
+        byte_array[pos / 2] = (uint8_t)(charmap[idx0] << 4) | charmap[idx1];
+    };
+}
+
+void _bytes_to_hex(const uint8_t *byte_array, char *hex, uint8_t max_len)
```

Comment:
use fmt?

Reply:
I thought of it but in fmt there's only the logic for one byte or for an array of bytes but reversed. This is not what I'm doing here. Maybe that would deserve a PR for fmt ?

Reply:
You could use just the one-byte version in a loop. ;)

Reply:
@kaspar030, updated to use the new `fmt` functions. Works like a charm!

Reply:
> Works like a charm!

That's the benefit of proper testing ;-)